### PR TITLE
prevent setting an account's name to a preexisting account name

### PR DESCRIPTION
**Pull request by nickpell**
fixes http://manage.dimagi.com/default.asp?155151#867373

existing accounts with the same name are unaffected

@benrudolph

### CODE REVIEW

File: corehq/apps/accounting/forms.py [hunk 1]
```diff
@@ -160,6 +160,18 @@ def __init__(self, account, *args, **kwargs):
             )
         )
 
+    def clean_name(self):
+        existing_names = set(account.name for account in BillingAccount.objects.all())
```
Comment (benrudolph) [inline corehq/apps/accounting/forms.py]: Not too big of a deal but could be more efficient to use `values` https://docs.djangoproject.com/en/1.7/ref/models/querysets/#django.db.models.query.QuerySet.values

`BillingAccount.objects.values('name')`

Comment (benrudolph) [inline corehq/apps/accounting/forms.py]: Or even 
```
if BillingAccount.objects.filter(name = cleaned_data['name']).exists() and self.account and (self.account.name != cleaned_data['name'])
  raise ValidationError("Name '%s' is already taken." % name)
```
might be even more efficient given there's an index on `name`. `all()` could return a lot of objects? but this operation might not be too frequent

File: corehq/apps/accounting/forms.py [hunk 2]
```diff
@@ -160,6 +160,18 @@ def __init__(self, account, *args, **kwargs):
...
+        if self.account:
+            disallowed_names = existing_names - {self.account.name}
+        else:
+            disallowed_names = existing_names
+
+        name = self.cleaned_data['name']
+        if name in disallowed_names:
+            raise ValidationError("Name '%s' is already taken." % name)
```
Comment (benrudolph) [inline corehq/apps/accounting/forms.py]: could `name`  potentially be unicode

Comment (nickpell) [inline corehq/apps/accounting/forms.py]: unlikely but possible, will fix

File: corehq/apps/accounting/forms.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/accounting/forms.py]: probably not a big deal but a good learning opportunity ... fixed

File: corehq/apps/accounting/forms.py (hunk 2 above)
Comment (nickpell) [inline corehq/apps/accounting/forms.py]: I actually think this code is unicode-safe.

Example:
```
"unicode value: %s" % u'ç'
u'unicode value: \xe7'
```

**Comment by benrudolph** [comment]
sweet, i'll just wait for build then merge

**Pull request merged**